Improve the playground

lydell opened the pull request:
- Add helpful buttons:
  - "Clear" to empty the code input.
  - "Copy link" for those who don't see that the URL changes.
  - "Copy markdown" to copy a markdown link and inline code snippets.
  - "Report issue" to open a new issue with the playground prefilled.
- Add "show second format (debug)" checkbox for debugging idempotency issues. 
- Update the issue template on build to be in sync with the "Report issue" button.
- Extra things I did while at it:
  - Change CodeMirror mode based on parser.
  - Fix some HTML errors and warnings.
  - Clean up CSS.
  - Minor design tweaks.
  - Updated the default example code.
  - And more...

Example "Copy as markdown" output:

**Prettier 1.6.1**
[Playground link](http://localhost:3000/playground/#%7B%22content%22%3A%22const%20Hello%20%3D%20(%7Bname%7D)%20%3D%3E%20%3Cdiv%20class%3D%5C%22HelloWorld%5C%22%20onClick%3D%7B()%20%3D%3E%20%7Balert(%5C%22Click!%5C%22)%7D%7D%3EHello%2C%20%3Cstrong%3E%7Bname%7D!%3C%2Fstrong%3E%3C%2Fdiv%3E%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A4%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Atrue%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D)
```
--jsx-bracket-same-line
--parser typescript
--single-quote
--tab-width 4
```

**Input:**
```tsx
const Hello = ({name}) => <div class="HelloWorld" onClick={() => {alert("Click!")}}>Hello, <strong>{name}!</strong></div>
```

**Output:**
```tsx
const Hello = ({ name }) => (
    <div
        class="HelloWorld"
        onClick={() => {
            alert('Click!');
        }}>
        Hello, <strong>{name}!</strong>
    </div>
);

```

ikatyang commented:
GitHub allowed to fill in the issue template from url query, e.g. https://github.com/prettier/prettier/issues/new?title=My+Title&body=My+Content, it'd be better to also add a <kbd>Report Issue</kbd> button so that user can create issue directly.

And I personally recommended to share the ISSUE_TEMPLATE generator function (using UMD?) so that our issue template will be always synchronized.

vjeux commented:
This is really awesome! I don't have the time to review it but feel free to merge if no one reviews it :)

azz commented on website/pages/playground/index.html:
```diff
@@ -246,41 +311,41 @@
         prettier version
         <span class="version"></span>
-        (master)
+        / master
```
The playground itself is redeployed but the version of prettier is not automatically rebuilt

lydell replied:
Is there a disadvantage to let the playground always use latest master? If so, I'll look into updating website/deploy.sh to also build Prettier.

azz replied:
Not that I can think of...

vjeux replied:
Yes, it's confusing for people to see master on the playground where they can't actually use it. We should have the last release by default and if we can manage, then having an option to display other versions.

We're going to do less drastic changes going forward so it may not be such a big deal anymore though.

lydell replied:
Ok, I'll leave it to use the latest release, then.

azz commented:
We should also add a "clear" button as it is quite annoying to select text on mobile.

vjeux commented:
I would have loved using it just now :)

lydell commented:
I've addressed all comments now – and added some more useful stuff. See the first comment. Unfortunately, I had to change indentation and move some stuff around in playground.js, so the diff is a bit messy.

![screenshot from 2017-09-14 00-58-54](https://user-images.githubusercontent.com/2142817/30404641-eb67cfaa-98e7-11e7-83a7-f9c497d8f159.png)

ikatyang commented on .github/ISSUE_TEMPLATE.md:
````diff
@@ -1,15 +1,28 @@
 
-**Expected behavior:**
+**Output:**
+```jsx
````
Can we have a blank line after the `**Title:**`? as it looks more comfortable. 

![image](https://user-images.githubusercontent.com/8341033/30406217-10130e1a-9923-11e7-88c3-31dcf5d06b5d.png)

lydell replied:
The markdown does look a bit tider, but:

- I don't like the extra spacing between "Prettier 1.6.1" and "Playground link" in the rendered HTML.
- It makes the issue template appear longer.

ikatyang replied:
Oh! I forgot it'll cause additional blank line in HTML.

azz commented on website/static/playground.js:
```diff
@@ -1,154 +1,313 @@
+        "",
+        "  if(!greeting){return null};",
+        "",
+        "     // TODO: Don't use random in render",
```
Sneaky :P

lydell commented on .github/ISSUE_TEMPLATE.md:
````diff
@@ -1,15 +1,28 @@
+**Prettier 1.6.1**
+[Playground link](https://prettier.io/playground/#.....)
+```
+# Options (if any):
````
I don't see any difference when I add `sh` or `bash`.

ikatyang replied:
![image](https://user-images.githubusercontent.com/8341033/30413710-4466b610-9952-11e7-841e-4c9446d47deb.png)

lydell replied:
Ah – you mean for the "comment". I didn't think of testing that, since the "Report issue" button never outputs any comments.

lydell commented on .github/ISSUE_TEMPLATE.md:
````diff
@@ -1,15 +1,28 @@
 
-```js
+**Prettier 1.6.1**
+[Playground link](https://prettier.io/playground/#.....)
````
I think it looks more cluttered that way.

lydell commented:
Updates:

- Added `sh` syntax highlighting.
- Added JSX syntax highlighting.

![screenshot from 2017-09-14 08-52-10](https://user-images.githubusercontent.com/2142817/30415859-3c7610e0-992a-11e7-8b6d-decc701d0046.png)

azz commented:
Looking good! Ready to merge?

vjeux commented:
> show second format (debug)

Soooo good, thanks for building that :)
